packages/desmos-std/src/msg_router.rs

```diff
@@ -0,0 +1,177 @@
+use crate::{
+    profiles::{
+        msg_router::ProfilesMsgRouter,
+        msg_routes::ProfilesMsgs,
+        msg_routes::ProfilesMsgs::{
+            AcceptDtagTransferRequest, BlockUser, CancelDtagTransferRequest, CreateRelationship,
+            DeleteProfile, DeleteRelationship, RefuseDtagTransferRequest, RequestDtagTransfer,
+            SaveProfile, UnblockUser,
+        },
+    },
+    types::DesmosRoute,
+};
+use cosmwasm_std::{Addr, CosmosMsg, CustomMsg};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct DesmosMsgRouter {
+    pub route: DesmosRoute,
+    pub msg_data: DesmosMsg,
+}
+
+impl Into<CosmosMsg<DesmosMsgRouter>> for DesmosMsgRouter {
+    fn into(self) -> CosmosMsg<DesmosMsgRouter> {
+        CosmosMsg::Custom(self)
+    }
+}
+
+impl CustomMsg for DesmosMsgRouter {}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[non_exhaustive] // missing chain-links and app-links messages (not necessary to me)
+pub enum DesmosMsg {
+    Profiles(ProfilesMsgs),
+}
+
+impl ProfilesMsgRouter<DesmosMsgRouter> for DesmosMsgRouter {
+    fn save_profile(
+        dtag: String,
+        creator: Addr,
+        nickname: Option<String>,
```

Comment:
I think Option should be remove here and checked by the the execute function which call `MsgRouter` api.

Reply:
I first did that thinking about the fact that those are optional fields but that not make much sense since they will later be saved as empty strings by the Desmos app.